### ISSUE TITLE
Is it possible to use custom or system ssh?

### ISSUE THREAD

**Issue by yangzhaofeng** [billziss-gh/sshfs-win#86]
On Unix-like systems, sshfs is seperate from ssh and updates much slower than ssh and ssl libraries. But the releases of this repository contain ssh binary that may not up-to-date. 

I tried to use the ssh provided by the windows system by substituting the ssh.exe in the install path with a symbolic link of C:\Windows\System32\OpenSSH\ssh.exe. However, it prompted "Access denied" when I reconnect to the sftp server(after entering my password).

**Comment by billziss-gh** [comment]
I do not test with the new Windows OpenSSH. It should work however with latest Cygwin SSH.

**Comment by shawnz** [comment]
Please add support for this functionality, otherwise it is not possible to use keys in the Windows SSH agent with sshfs. I have tried various methods to get it working and similarly faced errors. If this functionality could be added then this would become the perfect SFTP/SCP client for me.

**Comment by Ciantic** [comment]
Yes exactly, I'd like to have this as well because of the keys. 

There is already a good support structure building around the official `C:\Windows\System32\OpenSSH\ssh.exe`, among one of them is ability to read SSH keys automatically from KeePass (if you have KeeAgent). It's so good.

**Comment by shawnz** [comment]
@billziss-gh I'm sorry to be a bother but is there any way you could please take a look at this issue? If my understanding is correct then the different SSH clients should be drop-in replacements for each other, so I can't imagine the underlying issue here being too major. Yet this is incredibly important functionality for me which is basically preventing me from using this software currently.

If it turns out that the underlying issue is more complicated than I'm thinking, well then I'd understand if you don't want to patch it. But could you maybe just take a look in case it happens to be something small?

**Comment by billziss-gh** [comment]
@shawnz

My original objective with this project was to do a straight port of sshfs to Cygwin and WinFsp and this has been achieved. Later my objective became to incorporate these changes into the upstream SSHFS project and this has also been achieved.

Unfortunately I do not have the time to add features beyond what SSHFS already supports in the Cygwin environment. However SSHFS now compiles out of the box on Cygwin, so it is possible to take the SSHFS code, build it on Cygwin (requires FUSE for Cygwin, bundled with WinFsp) and support any special scenarios you want this way.

I am aware of one fork that claims support of OpenSSH by @sganis:

https://github.com/sganis/sshfs-win

**Comment by shawnz** [comment]
I don't really think this is a special scenario or an extra feature on top of what SSHFS is designed to provide. SSHFS doesn't claim to only be compatible with Cygwin's build of openSSH, and there's no obvious reason to me why it shouldn't work with Microsoft's build or any other. So I see this as a bug, not a feature request.

However I appreciate this alternate suggestion and I will be giving it a try, thank you for the link. And thank you for your work on this project.

**Comment by billziss-gh** [comment]
> SSHFS doesn't claim to only be compatible with Cygwin's build of openSSH, and there's no obvious reason to me why it shouldn't work with Microsoft's build or any other.

You may be correct on this. However the problem here is that OpenSSH (as included in Windows) is a Win32 application, while SSHFS (as included in this project) is a Cygwin application. For this reason I find it unlikely that they will interoperate well.

OTOH, I have not tried it so it is actually possible that they do.

**Comment by sganis** [comment]
Just to clarify a comment, I forked sshfs-win in an attempt to overcome the cygwin issues in a corporate environment, but I couldn’t go far, so I did a new implementation using libssh2 in windows without any cygwin dependency: https://github.com/sganis/golddrive

Sganis.

**Comment by mhogomchungu** [comment]
> Yes exactly, I'd like to have this as well because of the keys.
> 
> There is already a good support structure building around the official `C:\Windows\System32\OpenSSH\ssh.exe`, among one of them is ability to read SSH keys automatically from KeePass (if you have KeeAgent). It's so good.

sshfs-win can read SSH keys automatically from KeePass 2. I develop SiriKali[1] and sshfs-win is one of its backends on windows and reading keys automatically from KeePass 2 is a supported feature.

Are you sure you are not making a mistake somewhere?

[1] https://mhogomchungu.github.io/sirikali/

**Comment by bersbersbers** [comment]
@sganis is your fork usable? I don't see any Issues, Releases etc.
